### PR TITLE
Support SIGINT signal.

### LINKED ISSUE #175: CTRL-C should stop the query
When running a query that takes time, there is no way to stop it on the CLI. Pressing CTRL-C should stop the running query.

### PR DESCRIPTION
**Pull request by tzzed**
This fixes #175 .

Added `context.WithCancel` and passed it to `app.RunContext(ctx, os.Args)`.

### CODE REVIEW

**Review by asdine — COMMENTED** [review]
Thanks @tzzed ! Handling cancelation is more complex than that because we want to stop the query but not necessarily the program, and we also want to quit it properly.
I'm already working on a rewrite of the shell package to support that among other things, sorry I should have assigned the issue to me. I hope you learned a few things by working on this.

**Comment by tzzed** [comment]
it's ok for me. It was one of my interrogation ! should we close the `db` and `exit`.  I look forward your PR. ;)

**Review by tie — CHANGES_REQUESTED** [review]
@tzzed thanks, being able to cancel long-running operations from Genji shell is nice!
This would be really useful once we reach a consensus on context.Context API design.

**Comment by tie** [comment]
@tzzed @asdine I’ve left a few review comments even though look like this won’t get merged 🤗
Should we close this PR for now?

File: cmd/genji/main.go [hunk 1]
```diff
@@ -140,7 +143,31 @@ $ curl https://api.github.com/repos/genjidb/genji/issues | genji insert --db my.
...
+	ctx, cancel := context.WithCancel(context.Background())
+	app.Before = func(c *cli.Context) error {
+		sig := make(chan os.Signal, 1)
+		signal.Notify(sig, syscall.SIGINT)
+
+		go func() {
+			defer func() {
+				signal.Stop(sig)
+				cancel()
+			}()
```
[the remark this answers is not in the document]
Comment (tzzed) [reply cmd/genji/main.go:155]: I used it because we are inside ´App ´ configurator.

**Comment by asdine** [comment]
Yes let's close it 👍🏼

**Pull request closed without merging**